### ISSUE TITLE
Graviton 1 support

### ISSUE THREAD

**Issue by ivanbishop** [front-matter/invenio-rdm-starter#54]
Hi, this is a GREAT project, but I own and RI on AWS that is 1st gen Graviton with older crypto

So when I try and pull opensearch I see

Fatal glibc error: This version of Amazon Linux requires a newer ARM64 processor
search-1  |                    compliant with at least ARM architecture 8.2-a with Cryptographic
search-1  |                    extensions. On EC2 this is Graviton 2 or later.
search-1  | Fatal glibc error: This version of Amazon Linux requires a newer ARM64 processor
search-1  |                    compliant with at least ARM architecture 8.2-a with Cryptographic
search-1  |                    extensions. On EC2 this is Graviton 2 or later.
search-1  | Fatal glibc error: This version of Amazon Linux requires a newer ARM64 processor
search-1  |                    compliant with at least ARM architecture 8.2-a with Cryptographic
search-1  |                    extensions. On EC2 this is Graviton 2 or later.
db-1      | syncing data to disk ... ok

Is there any way you can build for Graviton1 also or are you hardwired to using HW features of the Graviton 2?

Thanks
Ivan

**Comment by mfenner** [comment]
@ivanbishop the opensearch docker image is provided by the opensearch community and not the inveniordm community. Ask for help there.

**Comment by ivanbishop** [comment]
Shall do, thanks Martin. 
If they have a graviton1, is it easy enough to make it part of the pull into your RDM offering? Via your compose file? 
thanks 
Ivan

**Comment by ivanbishop** [comment]
Note: OpenSearch images use [Amazon Linux 2⁠](https://aws.amazon.com/amazon-linux-2/) as the base image for 1.x versions and 2.x versions up until 2.9.0, and [Amazon Linux 2023⁠](https://aws.amazon.com/linux/amazon-linux-2023/) since 2.10.0.

I edited the docker compose file and simply told it to download opensearch 
_docker pull opensearchproject/opensearch:2.9.0_

This build is ghraviton1 capable

The whole build process per your instructions ran smoothly

Will look at fixing **allowed hosts** ? as I can't browse the UI as yet on 7080 (I altered it from port 80 as it was in use) and I see all the docker ports open and mapped per _docker ps -a_

**Comment by mfenner** [comment]
Are you using Docker Compose on a local computer or setting up infrastructure in AWS? For running Docker Compose locally you can override the settings, e.g. with an env file or another compose file.

For a local install you can set APP_ALLOWED_HOSTS = ['localhost', '127.0.0.1', '0.0.0.0']. In a production environment I would use a specific hostname.

**Comment by ivanbishop** [comment]
setting up on AWS Martin, I'll add in my DNS server name , its a global name,not in Route53 

Are there any specifics I should concern myself with for a "production" deploy? 

many thanks 
Ivan

**Comment by mfenner** [comment]
Using Docker Compose in a production environment with AWS is possible, but not typical. The invenio-rdm-starter Docker image can be safely used in production, but for container orchestration on AWS you would typically use ECS or EKS services. You can ask  other InvenioRDM project partners in the Invenio Discord Forum for their experience with AWS.

**Comment by ivanbishop** [comment]
Yeah, I do a lot of ECS/EKS work. I want to run Invenio past those who will use it in anger and see if they like it. (I do) If they do, then  I'll plan to move it to EKS

thanks Martin
